src/contexts/Set/getSetDetails.ts

```diff
@@ -0,0 +1,34 @@
+import { provider } from 'web3-core'
+import Set from "set.js"
+import { basicIssuanceModuleAddress, controllerAddress, debtIssuanceModuleAddress, navIssuanceModuleAddress, protocolViewerAddress, streamingFeeModuleAddress, setTokenCreatorAddress, tradeModuleAddress, masterOracleAddress, governanceModuleAddress, dpiTokenAddress, mviTokenAddress } from "../../constants/ethContractAddresses";
+import { SetDetails } from "set.js/dist/types/src/types";
+
+export async function getSetDetails(web3Provider: provider, productAddress: string): Promise<SetDetails[]> {
+  if (basicIssuanceModuleAddress === undefined ||
+      controllerAddress === undefined ||
+      masterOracleAddress === undefined ||
+      navIssuanceModuleAddress === undefined ||
+      protocolViewerAddress === undefined ||
+      setTokenCreatorAddress === undefined ||
+      streamingFeeModuleAddress === undefined ||
+      tradeModuleAddress === undefined ||
+      governanceModuleAddress === undefined ||
+      debtIssuanceModuleAddress === undefined) {
+      throw new Error("A set JS address is not defined. Please check your .env file")
+  }
+  const set = new Set({
+    web3Provider: web3Provider,
+    basicIssuanceModuleAddress: basicIssuanceModuleAddress,
+    controllerAddress: controllerAddress,
+    masterOracleAddress: masterOracleAddress,
+    navIssuanceModuleAddress: navIssuanceModuleAddress,
+    protocolViewerAddress: protocolViewerAddress,
+    setTokenCreatorAddress: setTokenCreatorAddress,
+    streamingFeeModuleAddress: streamingFeeModuleAddress,
+    tradeModuleAddress: tradeModuleAddress,
+    governanceModuleAddress: governanceModuleAddress,
+    debtIssuanceModuleAddress: debtIssuanceModuleAddress,
+  });
+
+  return set.setToken.batchFetchSetDetailsAsync([productAddress], []);
```

Comment:
I believe you can pass in modules as the second argument here to fetch related details. (Not that you would know that given our documentation...)

Anyways, here's a snippet from our tokensets repo:
```
    const moduleAddresses = [
      isPolygon
        ? process.env.POLYGON_SETJS_BASIC_ISSUANCE_MODULE
        : process.env.SETJS_BASIC_ISSUANCE_MODULE,
      isPolygon
        ? process.env.POLYGON_SETJS_STREAMING_FEE_MODULE
        : process.env.SETJS_STREAMING_FEE_MODULE,
      isPolygon ? process.env.POLYGON_SETJS_TRADE_MODULE : process.env.SETJS_TRADE_MODULE,
      isPolygon
        ? process.env.POLYGON_SETJS_DEBT_ISSUANCE_MODULE
        : process.env.SETJS_DEBT_ISSUANCE_MODULE,
    ];

    const setDetailsList: ISetDetails[] = await setJS.setToken.batchFetchSetDetailsAsync(
      addresses,
      moduleAddresses,
      caller,
    );
```

Reply:
Thanks for sharing! I'll use `moduleAddresses` when I fetch set details in a future PR